src/System.Text.Json/tests/Serialization/Value.ReadTests.cs

```diff
@@ -301,5 +295,38 @@ public static void ReadPrimitiveUri()
             uri = JsonSerializer.Deserialize<Uri>(@"""~/path""");
             Assert.Equal("~/path", uri.ToString());
         }
-   }
+
+        private static int SingleToInt32Bits(float value)
+        {
+#if BUILDING_INBOX_LIBRARY
+            return BitConverter.SingleToInt32Bits(value);
+#else
+            return Unsafe.As<float, int>(ref value);
+#endif
+        }
+
+        private static void AssertFloatingPointBehavior<T>(T netcoreExpectedValue, Func<T> testCode)
+        {
+            if (PlatformDetection.IsFullFramework)
+            {
+                Assert.Throws<JsonException>(() => testCode());
+            }
+            else
+            {
+                Assert.Equal(netcoreExpectedValue, testCode());
+            }
+        }
```

Comment:
Can we use existing helpers instead, e.g.
https://github.com/dotnet/corefx/blob/7094a5b94f4046abccfa2c1e6c9ab36243d8b3da/src/Common/tests/CoreFx.Private.TestUtilities/System/AssertExtensions.cs#L129
?

Reply:
This is a case where Full Framework is expected to throw and where .NET Core is expected to succeed. The behavior is very specific to these tests and serializing these floating-point values.